### PR TITLE
Added popular datasets on home page

### PR DESCRIPTION
**Pull request by matloob-altaf**
- Added most_popular_datasets() functions in plugin.py
- Added these datasets on the home page

### NOTE: HTML and CSS should be improved before merging...

### CODE REVIEW

**Comment by matloob-altaf** [comment]
Improved HTML and CSS with the help of Momin, now ready to merge

File: ckanext/ckan_theme/plugin.py [hunk 1]
```diff
@@ -1,13 +1,56 @@
 import ckan.plugins as plugins
 import ckan.plugins.toolkit as toolkit
 
+def most_popular_datasets():
+    '''
+    A function to return top 4 datasets based on popularity
+
+    '''
+    import requests
```
Comment (hussnainwithss) [inline ckanext/ckan_theme/plugin.py:9]: Place imports outside of the function

File: ckanext/ckan_theme/plugin.py [hunk 2]
```diff
@@ -1,13 +1,56 @@
...
+    import json
+    import socket 
+    
+    # Get the ip address   
+    hostname = socket.gethostname()    
+    ip_addr = socket.gethostbyname(hostname)
+    
+    # Make the HTTP request.
+    url = 'http://' + str(ip_addr) + '/api/3/action/package_search?sort=popularity+desc&rows=4'
+    response = requests.get(url)
```
Comment (hussnainwithss) [inline ckanext/ckan_theme/plugin.py:19]: use a try catch block for this

File: ckanext/ckan_theme/templates/home/index.html [hunk 3]
```diff
@@ -2,27 +2,78 @@
...
+    </div>
+  </div>
+</section> -->
+<section id="popular_dataset">
+  <div class="container">
+      <h1 id="pd_h">Featured Dataset</h1>
+      <div class="row pd_row">
+          <div class="col-6 text-left">
+              <div class="pd_heading">
+                  <h2>Popular Dataset</h2>
```
Comment (hussnainwithss) [inline ckanext/ckan_theme/templates/home/index.html:56]: Give this class of the page heading look at about.html page for reference

File: ckanext/ckan_theme/templates/home/index.html [hunk 4]
```diff
@@ -2,27 +2,78 @@
...
+        </a>
+        <a class="carousel-control right" href="#odp-main-slider" data-slide="next">
+          <span class="fa fa-angle-double-right"></span>
+        </a>
+      </div>
+    </div>
 </section>
 {{ super() }}
-{% endblock %}
+<!-- comments by hussnain put the section block under this line see the carousal section and odp page for reference -->
```
Comment (hussnainwithss) [inline ckanext/ckan_theme/templates/home/index.html:29]: This section needs to be inside the endblock. by placing this section for popular data sets outside the block you have excluded it from the primary contact block

File: ckanext/ckan_theme/templates/home/index.html [hunk 5]
```diff
@@ -2,27 +2,78 @@
...
+  <div class="container">
+      <h1 id="pd_h">Featured Dataset</h1>
+      <div class="row pd_row">
+          <div class="col-6 text-left">
+              <div class="pd_heading">
+                  <h2>Popular Dataset</h2>
+                  <p>Most Viewed Popular Dataset</p>
+              </div>
+              {% for data in h.ckan_theme_most_popular_datasets() %}
+              <div class="pd_datasets">
```
Comment (hussnainwithss) [inline ckanext/ckan_theme/templates/home/index.html:60]: improve this styling

File: ckanext/ckan_theme/templates/home/index.html [hunk 6]
```diff
@@ -2,27 +2,78 @@
...
+                  <div class="row">
+                      <div class="col">
+                          <h3><a href="/dataset/{{data['name']}}">{{data['title']}}</a></h3>
+                          <p>{{data['notes']}}</p>
```
Comment (hussnainwithss) [inline ckanext/ckan_theme/templates/home/index.html:64]: notes are not printing correctly its not formating text thek sy look why is that

File: ckanext/ckan_theme/templates/home/index.html [hunk 7]
```diff
@@ -2,27 +2,78 @@
...
+                          <ul class="list-inline">
+                            {% for resource in data['resources'] %}
+                              <li class="list-inline-item"><a href="/dataset/{{data['name']}}" data-format="{{resource['format']}}">{{resource['format']}}</a></li>
```
Comment (hussnainwithss) [inline ckanext/ckan_theme/templates/home/index.html:67]: replace this href by using url_for function of jinja2. look at [themeing guide](https://docs.ckan.org/en/2.8/theming/best-practices.html#use-url-for) for reference

**Review by hussnainwithss — CHANGES_REQUESTED** [review]

File: ckanext/ckan_theme/templates/home/index.html [hunk 8]
```diff
@@ -2,27 +2,78 @@
...
+    <ul>
+      {% for resource in data['resources'] %}
+      <li><a href="/dataset/{{data['name']}}" data-format="{{resource['format']}}">{{resource['format']}}</a></li>
+      {% endfor %}
+    </ul>
+    {% endfor %}
+    </div>
+  </div>
+</section> -->
+<section id="popular_dataset">
```
Comment (hussnainwithss) [inline ckanext/ckan_theme/templates/home/index.html:50]: Please use class for this kind of task as we might need to reuse these for another div say for popular organizations. in this case class makes more sense

**Review by hussnainwithss — CHANGES_REQUESTED** [review]

File: ckanext/ckan_theme/plugin.py (hunk 1 above)
Comment (matloob-altaf) [reply ckanext/ckan_theme/plugin.py]: Done

File: ckanext/ckan_theme/plugin.py (hunk 2 above)
Comment (matloob-altaf) [reply ckanext/ckan_theme/plugin.py]: Done

File: ckanext/ckan_theme/templates/home/index.html (hunk 4 above)
Comment (matloob-altaf) [reply ckanext/ckan_theme/templates/home/index.html:29]: done

File: ckanext/ckan_theme/templates/home/index.html (hunk 5 above)
Comment (matloob-altaf) [reply ckanext/ckan_theme/templates/home/index.html]: done

File: ckanext/ckan_theme/templates/home/index.html (hunk 6 above)
Comment (matloob-altaf) [reply ckanext/ckan_theme/templates/home/index.html]: done

File: ckanext/ckan_theme/templates/home/index.html (hunk 7 above)
Comment (matloob-altaf) [reply ckanext/ckan_theme/templates/home/index.html]: done

**Pull request closed without merging**